src/kboot_gpu.c

```diff
@@ -674,5 +687,54 @@ int dt_set_gpu(void *dt)
     if (firmware_set_fdt(dt, gpu, "apple,firmware-compat", compat))
         return -1;
 
+    size_t data_a_size, data_b_size, globals_size;
+    if (rust_gpu_initdata_size(compat->num[0], compat->num[1], &data_a_size, &data_b_size,
+                               &globals_size) == -1)
+        return -1;
+
+    void *data_a = calloc(1, data_a_size);
+    void *data_b = calloc(1, data_b_size);
+    void *globals = calloc(1, globals_size);
+    size_t n_perf_states_cs = 0, n_perf_states_afr = 0;
+    const struct aux_perf_state *pstate_cs_raw = NULL, *pstate_afr_raw = NULL;
+    u64 uat_ttb_base;
+    ADT_GETPROP(adt, sgx, "gpu-region-base", &uat_ttb_base);
+    u64 gpu_base;
+
+    int adt_sgx_path[8];
+    if (adt_path_offset_trace(adt, "/arm-io/sgx", adt_sgx_path) < 0)
+        bail("ADT: GPU: Failed to get sgx\n");
+
+    if (adt_get_reg(adt, adt_sgx_path, "reg", 0, &gpu_base, NULL) < 0)
+        bail("ADT: GPU: Failed to get sgx reg 0\n");
+
+    u32 base_pstate;
+    if (ADT_GETPROP(adt, sgx, "gpu-perf-base-pstate", &base_pstate) < 0)
```

Comment:
this property does not exists on all devices. It is at least missing on M1 mac mini and imac